webhook/resources/volume/validator.go

```diff
@@ -452,3 +473,54 @@ func (v *volumeValidator) canDisableRevisionCounter(image string, dataEngine lon
 
 	return true, nil
 }
+
+func validateSnapshotMaxCount(snapshotMaxCount int) error {
+	if snapshotMaxCount < 2 || snapshotMaxCount > 250 {
+		return fmt.Errorf("snapshot max count should be between 2 to 250")
+	}
+	return nil
+}
+
+func validateSnapshotMaxSize(size, snapshotMaxSize int64) error {
+	if snapshotMaxSize != 0 && snapshotMaxSize < size*2 {
+		return fmt.Errorf("snapshot max size can not be 0 and at least twice of volume size")
+	}
+	return nil
+}
+
+func (v *volumeValidator) validateUpdatingSnapshotMaxCountAndSize(newVolume *longhorn.Volume) error {
+	var (
+		currentSnapshotCount     int
+		currentTotalSnapshotSize int64
+		engine                   *longhorn.Engine
+	)
+	engines, err := v.ds.ListVolumeEngines(newVolume.Name)
+	if err != nil && !datastore.ErrorIsNotFound(err) {
+		return werror.NewInternalError(fmt.Sprintf("can't list engines for volume %s, err %v", newVolume.Name, err))
+	} else if len(engines) >= 2 {
+		return werror.NewInvalidError("can't update snapshotMaxCount or snapshotMaxSize during migration", "")
+	} else if len(engines) == 0 {
+		return nil
+	}
+
+	for _, e := range engines {
+		engine = e
+	}
```

Comment:
engine := engines[0]

Reply:
The `engines` is map here, so we can't use `engines[0]`.